src/modules/Form/index.tsx

```diff
@@ -0,0 +1,199 @@
+        errorRef: createRef(),
+      },
+      avatarField: {
+        inputRef: createRef(),
+        errorRef: createRef(),
+      },
+      typeField: {
+        selectRef: createRef(),
+        errorRef: createRef(),
+      },
+      nameField: {
+        inputRef: createRef(),
+        errorRef: createRef(),
+      },
+    };
+  }
+
+  clearError = (errorRef: RefObject<HTMLSpanElement>) => {
+    if (this.state.initialEnter) this.setState({ initialEnter: false });
+    if (errorRef.current?.style.display === 'block') {
+      this.setState({ totalErrors: this.state.totalErrors - 1 });
+      errorRef.current.style.display = 'none';
+    }
+  };
+
+  clearFields = () => {
+    const {
+      birthdayField,
+      isShinyField,
+      consentField,
+      genderField,
+      avatarField,
+      nameField,
+      typeField,
+    } = this.state;
+    avatarField.inputRef.current && (avatarField.inputRef.current.value = '');
+    birthdayField.inputRef.current && (birthdayField.inputRef.current.value = '');
+    consentField.inputRef.current && (consentField.inputRef.current.checked = false);
+    genderField.maleInputRef.current && (genderField.maleInputRef.current.checked = false);
+    genderField.femaleInputRef.current && (genderField.femaleInputRef.current.checked = false);
+    isShinyField.inputRef.current && (isShinyField.inputRef.current.checked = false);
+    typeField.selectRef.current && (typeField.selectRef.current.value = '');
+    nameField.inputRef.current && (nameField.inputRef.current.value = '');
+  };
+
+  validateGenderField = () => {
+    const { errorRef, femaleInputRef, maleInputRef } = this.state.genderField;
+    if (femaleInputRef.current?.checked) return 'female';
+    else if (maleInputRef.current?.checked) return 'male';
+    else errorRef?.current && (errorRef.current.style.display = 'block');
+  };
+
+  validateConsentField = () => {
+    const { errorRef, inputRef } = this.state.consentField;
+    if (inputRef?.current?.checked) return inputRef.current.checked;
+    else errorRef?.current && (errorRef.current.style.display = 'block');
+  };
+
+  validateCommonField = ({ errorRef, inputRef }: CommonFieldType) => {
+    if (inputRef.current?.value && inputRef.current?.value.length > 2) {
+      return inputRef.current.value;
+    } else errorRef?.current && (errorRef.current.style.display = 'block');
+  };
+
+  validateTypeField = () => {
+    const { errorRef, selectRef } = this.state.typeField;
+    if (selectRef.current?.value) return selectRef.current.value;
+    else errorRef?.current && (errorRef.current.style.display = 'block');
+  };
+
+  handleSubmit = (event: FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    const type = this.validateTypeField();
+    const gender = this.validateGenderField();
+    const consent = this.validateConsentField();
+    const name = this.validateCommonField(this.state.nameField);
+    const birthday = this.validateCommonField(this.state.birthdayField);
```

Comment:
in addition to the remark above:
Will you call the validation of a million fields one at a time?